tools/topology/sof-tgl-rt711-rt1308.m4

```diff
@@ -20,22 +20,49 @@ include(`common/tlv.m4')
 # Include Token library
 include(`sof/tokens.m4')
 
-include(`platform/intel/tgl.m4')
+include(`platform/intel/'PLATFORM`.m4')
 
 # Define pipeline id for intel-generic-dmic.m4
 # to generate dmic setting
-
-ifelse(CHANNELS, `0', ,
+# If there is external amplifier connected to the SDW codec rt1308,
+# enable "EXT_AMP" option in the CMakefile.
+ifdef(`EXT_AMP',
 `
-define(DMIC_PCM_48k_ID, `10')
-define(DMIC_PIPELINE_48k_ID, `4')
-define(DMIC_DAI_LINK_48k_ID, `3')
-
-define(DMIC_PCM_16k_ID, `11')
-define(DMIC_PIPELINE_16k_ID, `5')
-define(DMIC_DAI_LINK_16k_ID, `4')
+ ifdef(`CHANNELS',
+ `
+  define(HDMI_BELINK_ID, `5')
+
+  ifdef(`DMICPROC', , `define(DMICPROC, eq-iir-volume)')
+  ifdef(`DMIC16KPROC', , `define(DMIC16KPROC, eq-iir-volume)')
+  define(DMIC_PCM_48k_ID, `10')
+  define(DMIC_PCM_16k_ID, `11')
+  define(DMIC_DAI_LINK_48k_ID, `3')
+  define(DMIC_DAI_LINK_16k_ID, `4')
+  define(DMIC_PIPELINE_48k_ID, `4')
+  define(DMIC_PIPELINE_16k_ID, `5')
+  include(`platform/intel/intel-generic-dmic.m4')
+  ',
+ `define(HDMI_BELINK_ID, `3')'
+ )
+',
 
-include(`platform/intel/intel-generic-dmic.m4')
+`
+ ifdef(`CHANNELS',
+ `
+  define(HDMI_BELINK_ID, `4')
+
+  ifdef(`DMICPROC', , `define(DMICPROC, eq-iir-volume)')
+  ifdef(`DMIC16KPROC', , `define(DMIC16KPROC, eq-iir-volume)')
+  define(DMIC_PCM_48k_ID, `10')
+  define(DMIC_PCM_16k_ID, `11')
+  define(DMIC_DAI_LINK_48k_ID, `2')
+  define(DMIC_DAI_LINK_16k_ID, `3')
+  define(DMIC_PIPELINE_48k_ID, `4')
+  define(DMIC_PIPELINE_16k_ID, `5')
```

Comment:
Sorry @libinyang this can be simplified quite a bit. there is no reason for duplication of the DMIC_PIPELINE_ID, DMIC_PCM_ID and DMICPROC. 

You only need to change the BE ID.

nested definitions are really hard to track with M4, so you want to make them as simple as possible. thanks!